assets/app.js

```diff
@@ -794,39 +825,54 @@ new Vue({
                 }
-                deleted_resource_name = product.deleted_materials_resources.map(function (resource) {
-                    return resource.name
-                });
-                if (deleted_resource_name.length === 0) {
-                    return true
+
+                if (type == 'component1' && product['is_rejected'] && product['rejected_component1']) {
+                    if (product[type] == null || Object.keys(product[type]).length === 0) {
+                        // there is component2 but no component1
+                        verified_resources.component1 = (product['component2'] == null || Object.keys(product['component2']).length === 0 || product['component2'].name === '-')
+                    }
+                        // component1 was deleted
+                    if (product[type] !== null && Object.keys(product[type]).length > 0 && product[type].deleted) {
+                        verified_resources.component1 = false
+                    }
                 }
-                resource_names = product[type].map(function (resource) {
-                    return resource.name
-                })
-                return deleted_resource_name.find(function (name) {
-                    return !(resource_names.indexOf(name) > -1)
-                });
-            }
 
-            if (type == 'component1' && product['is_rejected'] && product['rejected_' + type]) {
-                if (product[type] == null || Object.keys(product[type]).length === 0 || product.length === 0) {
-                    return (product['component2'] == null || Object.keys(product['component2']).length === 0 || product.length === 0 || product['component2'].name === '-')
+                if (type === 'component2' && product['is_rejected'] && product['rejected_component1']) {
+                    // component2 was deleted
+                    if (product[type] !== null && Object.keys(product[type]).length > 0 && product[type].deleted) {
+                        verified_resources.component2 = false
+                    }
                 }
-                return !(Object.keys(product['component1'])[0] === 'deleted_component1')
-            }
 
-            return !(product['is_rejected']
-            && product['rejected_' + type]
-            && (product[type] == null || Object.keys(product[type]).length === 0 || product.length === 0))
+                if (type == 'materials' && product['is_rejected'] && product['rejected_materials']) {
+                    if (product[type].length === 0) {
+                        verified_resources.materials = false
+                    }
+                    if (product.deleted_materials_resources.length > 0 && product.initial_materials_amount > 0) {
+                        product.dirty = true
+                        verified_resources['no_deleted_materials'] = false
+                    }
+                }
+            }
+            return verified_resources
```

Comment:
all the part above needs refactoring!
just a couple of things

-  don't repeat yourself! you check 5 times if one product is rejected
- if you know that `type ==materials/component/base_product` in a if condition please use it: `product['materials']` is then better then `product[type]`: easier to read and slightly faster